src/librustc_metadata/schema.rs

```diff
@@ -289,7 +289,7 @@ impl_stable_hash_for!(struct Entry<'tcx> {
 
 #[derive(Copy, Clone, RustcEncodable, RustcDecodable)]
 pub enum EntryKind<'tcx> {
-    Const(u8),
+    Const(u8, Lazy<RenderedConst>),
```

Comment:
You could replace the `u8` here and in `AssociatedConst` with a `struct ConstQualif { mir: u8, ast_promotable: bool }`.

Reply:
What does `ast_promotable` mean?

Reply:
The `rvalue_promotable_to_static: bool` field that's currently in `Ast`.